tuyaha/devices/base.py

```diff
@@ -36,15 +45,80 @@ def available(self):
     def iconurl(self):
         return self.icon
 
-    def update(self):
+    def _update_data(self, key, value, force_val=False):
+        if self.data:
+            if not force_val and self.data.get(key) is None:
+                return
+            self.data[key] = value
+            self.api.update_device_data(self.obj_id, self.data)
+
+    def _control_device(self, action, param=None):
+        success, response = self.api.device_control(self.obj_id, action, param)
+        if not success:
+            self._update_data("online", False)
+        else:
+            self._last_update = datetime.now()
+        return success
+
+    def _update(self, use_discovery):
```

Comment:
So previously we were using discovery only for switch, now we use it for any device type, right? And it doesn't cause backward-compatibility problems?
Why do we need this `use_discovery` then? You can safely remove this flag and QueryDevice mechanism.

Reply:
This is long story. I did a lot of test, API limit currently are:
- discovery can be called once every 300 seconds
- query can be called once every 60 seconds (at all, not for device)

I put this flag to have the option to choose the call to do depending on number of device registered, and also hoping in a change Tuya side, allowind to choose the best method without having to change the library